### ISSUE TITLE
Process not working

### ISSUE THREAD

**Issue by fcabrera23** [rdanter/wrl-rpi-bsp#2]
Hi, im trying to follow your tutorial, however every time I try to run setup, i get the following error.

bitbake wrlinux-image-std

NOTE: Fetching uninative binary shim from file:///home/parallels/my-project/layers/wrlinux/files/uninative-tarballs/a09922172c3a439105e0ae6b943daad2d83505b17da0aba97961ff433b8c21ab/x86_64-nativesdk-libc.tar.xz;sha256sum=a09922172c3a439105e0ae6b943daad2d83505b17da0aba97961ff433b8c21ab
Initialising tasks: 100% |####################################################################################################| Time: 0:00:02
Sstate summary: Wanted 1613 Found 0 Missed 1613 Current 0 (0% match, 0% complete)
NOTE: Executing Tasks
NOTE: Setscene tasks completed
ERROR: quilt-native-0.66-r0 do_fetch: Network access disabled through BB_NO_NETWORK (or set indirectly due to use of BB_FETCH_PREMIRRORONLY) but access requested with command /usr/bin/env wget -t 2 -T 30 --passive-ftp --no-check-certificate -P /home/parallels/my-project/build/downloads 'https://download.savannah.gnu.org/releases/quilt/quilt-0.66.tar.gz' (for url https://download.savannah.gnu.org/releases/quilt/quilt-0.66.tar.gz)
ERROR: Logfile of failure stored in: /home/parallels/my-project/build/tmp-glibc/work/x86_64-linux/quilt-native/0.66-r0/temp/log.do_fetch.8946
ERROR: Task (/home/parallels/my-project/layers/oe-core/meta/recipes-devtools/quilt/quilt-native_0.66.bb:do_fetch) failed with exit code '1'
ERROR: gnu-config-native-20190501+gitAUTOINC+b98424c249-r0 do_fetch: Network access disabled through BB_NO_NETWORK (or set indirectly due to use of BB_FETCH_PREMIRRORONLY) but access requested with command LANG=C git -c core.fsyncobjectfiles=0 clone --bare --mirror git://git.savannah.gnu.org/config.git /home/parallels/my-project/build/downloads/git2/git.savannah.gnu.org.config.git --progress (for url git://git.savannah.gnu.org/config.git)
ERROR: Logfile of failure stored in: /home/parallels/my-project/build/tmp-glibc/work/x86_64-linux/gnu-config-native/20190501+gitAUTOINC+b98424c249-r0/temp/log.do_fetch.8947
ERROR: Task (virtual:native:/home/parallels/my-project/layers/oe-core/meta/recipes-devtools/gnu-config/gnu-config_git.bb:do_fetch) failed with exit code '1'
ERROR: m4-native-1.4.18-r0 do_fetch: Network access disabled through BB_NO_NETWORK (or set indirectly due to use of BB_FETCH_PREMIRRORONLY) but access requested with command /usr/bin/env wget -t 2 -T 30 --passive-ftp --no-check-certificate -P /home/parallels/my-project/build/downloads 'https://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz' (for url https://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz)
ERROR: Logfile of failure stored in: /home/parallels/my-project/build/tmp-glibc/work/x86_64-linux/m4-native/1.4.18-r0/temp/log.do_fetch.8949
ERROR: Task (/home/parallels/my-project/layers/oe-core/meta/recipes-devtools/m4/m4-native_1.4.18.bb:do_fetch) failed with exit code '1'
NOTE: Tasks Summary: Attempted 5 tasks of which 0 didn't need to be rerun and 3 failed.

@rdanter  could you please verify if this tutorial is still working?

**Comment by fcabrera23** [comment]
@rdanter  i changed the BB_NO_NETWORK = 0 and now im getting the following error

Sstate summary: Wanted 1607 Found 0 Missed 1607 Current 64 (0% match, 3% complete)
NOTE: Executing Tasks
NOTE: Setscene tasks completed
WARNING: linux-yocto-4.19.87+gitAUTOINC+4f5d761316_b44ad1b1e7-r0 do_fetch: Failed to fetch URL git:///home/parallels/my-project/layers/wrlinux/git/linux-yocto.git;protocol=file;name=machine;branch=v4.19/standard/base;, attempting MIRRORS if available
ERROR: linux-yocto-4.19.87+gitAUTOINC+4f5d761316_b44ad1b1e7-r0 do_fetch: Fetcher failure: Unable to find revision b44ad1b1e7c685e75b7788a026a2416edc2ee656 in branch v4.19/standard/base even from upstream
ERROR: linux-yocto-4.19.87+gitAUTOINC+4f5d761316_b44ad1b1e7-r0 do_fetch: Fetcher failure for URL: 'git:///home/parallels/my-project/layers/wrlinux/git/linux-yocto.git;protocol=file;name=machine;branch=v4.19/standard/base;'. Unable to fetch URL from any source.
ERROR: Logfile of failure stored in: /home/parallels/my-project/build/tmp-glibc/work/qemux86_64-wrs-linux/linux-yocto/4.19.87+gitAUTOINC+4f5d761316_b44ad1b1e7-r0/temp/log.do_fetch.14026
ERROR: Task (/home/parallels/my-project/layers/oe-core/meta/recipes-kernel/linux/linux-yocto_4.19.bb:do_fetch) failed with exit code '1'
NOTE: Tasks Summary: Attempted 578 tasks of which 550 didn't need to be rerun and 1 failed.

**Comment by rdanter** [comment]
Hi @fcabrera23 

What is your git clone command line to get the wrlinux-x/setup.sh script?
How do you then run setup.sh and what is you bitbake command exactly?

The layer I have is for official WR Linux, and may not work with an unofficial release. The errors you are showing though don't seem related to the RPi BSP so you would probably get them with any BSP. Have you tried building qemux86-64 for example? Do you get the same error there?

Thanks
Rich

**Comment by fcabrera23** [comment]
Hi @rdanter  thanks for you answer. Do you have the image already compiled for rpi3-64?

For wrlinux-x I cloned from https://github.com/WindRiver-Labs/wrlinux-x
For the commands, i used your guidance:

$ git clone --branch WRLINUX_10_19_LTS /path/to/wrlinux-x
$ ./wrlinux-x/setup.sh --distro wrlinux --dl-layers --accept-eula yes
$ . environment-setup-x86_64-wrlinuxsdk-linux
$ . oe-init-build-env
$ bitbake-layers add-layer /path/to/wrl-rpi-bsp/

Thanks

**Comment by rdanter** [comment]
Hi @fcabrera23 

I have not tried the WindRiver-Labs version for a long time, let me give it a try today and I'll get back to you.

Rich

**Comment by rdanter** [comment]
Hi @fcabrera23 

I successfully built using the Labs version. One thing to note is that this version does not have the WRLINUX_10_19_LTS branch, so I used the default branch as follows:

```
$ git clone https://github.com/WindRiver-Labs/wrlinux-x
$ ./wrlinux-x/setup.sh --distro wrlinux --dl-layers --accept-eula yes
$ . environment-setup-x86_64-wrlinuxsdk-linux
$ . oe-init-build-env
$ bitbake-layers add-layer /path/to/wrl-rpi-bsp/
```

Add the following to conf/local.conf:

```
BB_NO_NETWORK = '0'
MACHINE = "rpi3-64"
```

Then:

```
$ bitbake wrlinux-image-std
```

So I think the only problem was the branch, which is relevant for the commercial version of WR Linux LTS 19, but not for the Labs version.

You will notice that the do_kernel_metadata build task will take a long time. This is something I reported and is fixed in the commercial version, but not yet in the Labs version. It should be resolved at some point though.

Rich
